### PR TITLE
WeightInfo for Identity Pallet

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR introduces the `WeightInfo` trait for the Identity Pallet, taking advantage of the automated benchmarking system.

Changes:

* Split `set_subs` benchmark into `set_subs_new` and `set_subs_old`.

This was needed because old subs and new subs both write to storage, and it was not possible to decouple the storage read/writes from one another. So I created two benchmarks, one which takes into account only adding new subs, and the other that takes into account only removing old subs.

This has known issues that the base weight and some base reads/writes are repeated, but I can't think of a cleaner solution given the current tooling. In general, the `set_subs` function looks to me to be very unoptimized and probably needs to be refactored all-together. For now, this weight is high, but safe.

* Fix benchmarks for `kill_identity` making sure to **not** whitelist the target account.
* Changed benchmark variable from `p` to `s` for add/rename/remove/quit_sub. Moved to bottom to match extrinsic order.
* Moved Identity tests to their own file.
* Integrated new weight formulas into pallet.
* Outputted benchmark files now use underscore to delimit large numbers.

### CODE REVIEW

File: frame/identity/src/lib.rs [hunk 1]
```diff
@@ -820,10 +651,15 @@ decl_module! {
...
-		)]
+		// TODO: This whole extrinsic screams "not optimized". For example we could
+		// filter any overlap between new and old subs, and avoid reading/writing
+		// to those values... We could also ideally avoid needing to write to
+		// N storage items for N sub accounts. Right now the weight on this function
+		// is a large overestimate due to the fact that it could potentially write
+		// to 2 x T::MaxSubAccounts::get().
+		#[weight = T::WeightInfo::set_subs_old(T::MaxSubAccounts::get()) // P: Assume max sub accounts removed.
+			.saturating_add(T::WeightInfo::set_subs_new(subs.len() as u32)) // S: Assume all subs are new.
+		]
```
Comment (shawntabrizi) [inline frame/identity/src/lib.rs:662]: Please check this is sane to others.

**Review by thiolliere — DISMISSED** [review]
looks good to me, the pointed optimisation makes sense to me, also some function could make use of refund such as remove_sub but it should be done in another PR

**Comment by thiolliere** [comment]
also don't you want to add generated weight_info in node-runtime as well ?

**Comment by shawntabrizi** [comment]
@thiolliere yes I will add everything to polkadot, probably as one larger pr with a few of these seperate weightinfos we have been adding.

**Review by thiolliere — APPROVED** [review]
ok if polkadot and node-template get their file updated in another PR then this one is ready

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**